src/api/services/participantsService.ts

```diff
@@ -105,12 +109,18 @@ const idParser = z.object({
 
 const hasParticipantAccess = async (req: ParticipantRequest, res: Response, next: NextFunction) => {
   const { participantId } = idParser.parse(req.params);
-  const participant = await Participant.query().findById(participantId);
+  const participant = await Participant.query().findById(participantId).withGraphFetched('types');
   if (!participant) {
     return res.status(404).send([{ message: 'The participant cannot be found.' }]);
   }
 
-  if (!(await isUserBelongsToParticipant(req.auth?.payload?.email as string, participantId))) {
+  const currentUserEmail = req.auth?.payload?.email as string;
+  if (
+    !(
+      (await isUserAnApprover(currentUserEmail)) ||
```

Comment:
If the user is an approver, bypass this isUserBelongsToParticipant check, the type of participant and approvable type will be checked in isApproverCheck middleware

Reply:
I'm pretty worried about this - won't this mean an approver could do *anything* to a participant? An approver should be able to approve an account, but they shouldn't for example be able to change sharing settings for that participant.